docs/csharp/whats-new/csharp-9.md

```diff
@@ -0,0 +1,237 @@
+
+You can also seal records to prevent further derivation:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/RecordsExamples.cs" ID="SealedRecord":::
+
+The compiler synthesizes different versions of the methods above. The method signatures depend on if the record type is sealed and if the direct base class is object. Records should have the following capabilities:
+
+- Equality is value-based, and includes a check that the types match. For example, a `Student` can't be equal to a `Person`, even if the two records share the same name.
+- Records have a consistent string representation generated for you.
+- Records support copy construction. Correct copy construction must include inheritance hierarchies, and properties added by developers.
+- Records can be copied with modification. These copy and modify operations supports non-destructive mutation.
+- All records support deconstruction.
+
+In addition to the familiar `Equals` overloads, `operator ==`, and `operator !=`, the compiler synthesizes a new `EqualityContract` property. The property returns a `Type` object that matches the type of the record. If the base type is `object`, the property is `virtual`. If the base type is another record type, the property is an `override`. If the record type is `sealed`, the property is `sealed`. The synthesized `GetHashCode` uses the `GetHashCode` from all the public properties declared in the base type and the record type. These synthesized methods enforce value-based equality throughout an inheritance hierarchy. That means a `Student` will never be considered equal to a `Person` with the same name. The types of the two records must match as well as all properties shared among the record types being equal.
+
+Records also have a synthesized constructor and a "clone" method for creating copies. The synthesized constructor has one argument of the record type. It produces a new record with the same values for all properties of the record. This constructor is private if the record is sealed, otherwise it's protected. The synthesized "clone" method supports copy construction for record hierarchies. The term "clone" is in quotes because the actual name is compiler generated. You can't create a method named `Clone` in a record type. The synthesized "clone" method returns the type of record being copied using virtual dispatch. If a record type is `abstract`, the clone method is also `abstract`. If a record type is `sealed` and its base type isn't `object`, the clone method is `sealed`. If the base type of the record is `object`, the clone method is `virtual` if the record type isn't `sealed`. Otherwise, it's `override`. The result of all these rules is the equality is implemented consistently across any hierarchy of record types. Two records are equal to each other if their properties are equal and their types are the same, as shown in the following example:
```

Comment:
> If the base type of the record is `object`, the clone method is `virtual` if the record type isn't `sealed`. Otherwise, it's `override`.

Does "otherwise" refer to the first "if" (base type is object), or the second if "record type isn't sealed".

For a sealed record whose base type is object, I'm getting Clone as only public.
The sentence in all cases seems to contradict with this, I think?

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABAzAAijggBs4V8E4BjAewQoAVEpaA7AWACgBvbgXyA

Adding a repro this time :smile:

Reply:
Good catch. Let me re-write this set of sentences. (I think it is just too close to the spec language, and isn't strict enough)